crates/syntax/src/ast/make.rs

```diff
@@ -430,6 +430,19 @@ pub fn match_arm(pats: impl IntoIterator<Item = ast::Pat>, expr: ast::Expr) -> a
     }
 }
 
+pub fn match_arm_with_guard(
+    pats: impl IntoIterator<Item = ast::Pat>,
+    guard: ast::Expr,
```

Comment:
I’d rather add `guard: Option<ast::Expr>` to the previous function. `make` is supposed to be maximally orthogonal.

Semi-related: saw this cute thing recently: https://roslynquoter.azurewebsites.net/

Reply:
Ye, makes sense.

That looks like a neat tool, on that note I should probably revisit the auto-generation for the make api now that I think about it 😄.